### PR TITLE
Fix user delete

### PR DESCRIPTION
**Pull request by maxlath**
cleaning to prevent a 500 error because of a [missing `comments_` variable](https://github.com/inventaire/inventaire/blob/bf45de0/server/controllers/comments/lib/comments.coffee)

### CODE REVIEW

File: tests/api/user/delete.test.coffee [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    .then (user)->
+      customAuthReq userPromise, 'delete', '/api/user'
+      .then (res)->
+        res.ok.should.be.true()
+        getRefreshedUser userPromise
+        .then (deletedUser)->
+          deletedUser._id.should.equal user._id
+          previousRevInteger = parseInt(user._rev .split('-')[0])
+          parseInt(deletedUser._rev.split('-')[0]).should.equal(previousRevInteger + 1)
+          deletedUser.username.should.equal user.username
```
Comment (jum-s) [inline tests/api/user/delete.test.coffee]: keeping username slot occupied ? what about i delete my account and someoneelse want to user de username i used

File: server/controllers/items/lib/delete_user_items.coffee [hunk 2]
```diff
@@ -6,16 +6,5 @@ items_ = __.require 'controllers', 'items/lib/items'
 promises_ = __.require 'lib', 'promises'
 
 module.exports = (userId)->
-  # get user items ids
   items_.byOwner userId
-  .then deleteItemsAndComments.bind(null, userId)
```
Comment (jum-s) [inline server/controllers/items/lib/delete_user_items.coffee]: what about delete comments too?

File: tests/api/user/delete.test.coffee (hunk 1 above)
Comment (maxlath) [reply tests/api/user/delete.test.coffee]: it's not possible at the moment, neither before or after this PR, but having a look at it and the implications could be a todo

File: server/controllers/items/lib/delete_user_items.coffee (hunk 2 above)
Comment (maxlath) [reply server/controllers/items/lib/delete_user_items.coffee]: not sure of what you mean

Comment (jum-s) [reply server/controllers/items/lib/delete_user_items.coffee]: garbage collect orphan comments

Comment (maxlath) [reply server/controllers/items/lib/delete_user_items.coffee]: there are no more items comments in the app, trying to delete those was thus legacy code that this PR removes

Comment (jum-s) [reply server/controllers/items/lib/delete_user_items.coffee]: ok

**Pull request merged**